packages/passwordless-auth-construct/src/types.ts

```diff
@@ -170,29 +272,16 @@ export type CustomAuthTriggers = {
  * SNS Service Configuration.
  */
 export type SnsServiceConfig = {
-  /**
-   * The origination number to use for SMS messages. Defaults to undefined.
-   */
-  originationNumber?: string;
-  /**
-   * The sender id to use for SMS messages. Defaults to undefined.
-   */
-  senderId?: string;
+  otp: Partial<SmsOptions>;
+  magicLink: Partial<SmsOptions>;
```

Comment:
question: why use `Partial` here?

Reply:
This was because the values won't be defined if otp or magicLink is disabled. This is a bit of a hack though. It makes us have to null check everything later. Thanks for drawing attention to this.

I made a few changes:
- I updated this to be a nullable type instead, which is what it should have been.
- I added new env vars to make it explicit if otp/magic link via sms/email is enabled (opposed to checking to see if certain env vars exist and assuming it is enabled).
- I added new types which reflect the env vars we expect for otp/magic link if they are enabled